homeassistant/components/esphome/sensor.py

```diff
@@ -71,8 +73,17 @@ def state(self) -> Optional[str]:
     @property
     def unit_of_measurement(self) -> str:
         """Return the unit the value is expressed in."""
+        if self._static_info.unit_of_measurement == "":
+            return None
         return self._static_info.unit_of_measurement
 
+    @property
+    def device_class(self) -> str:
+        """Return the class of this device, from component DEVICE_CLASSES."""
+        if self._static_info.device_class == "":
+            return None
+        return self._static_info.device_class
```

Comment:
The `device_class` entity property is a Home Assistant API so we need to validate that the returned value is valid. We don't trust values coming directly from a library. We can use this list:

https://github.com/home-assistant/core/blob/efb172cedd8a54a2a508e27b4110e3b7f6dc7bbd/homeassistant/components/sensor/__init__.py#L37-L50

Reply:
Thanks @MartinHjelmare, then I'll make a new pr for that with something like:
```python
        if not self._static_info.device_class in DEVICE_CLASSES :
            return None
```

Reply:
That's perfect. Thanks!